### PR TITLE
Fix missing transform import for `createLucideIcon` from lucide-react

### LINKED ISSUE #53605: `createLucideIcon` from `lucite-react` not working from v13.4.13-canary.5
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 23.0.0: Tue Jul 18 20:35:32 PDT 2023; root:xnu-10002.0.199.505.1~3/RELEASE_ARM64_T8112
    Binaries:
      Node: 20.3.1
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: 8.6.11
    Relevant Packages:
      next: 13.4.13-canary.16
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/joulev/debug/tree/nextjs-create-lucide-icon-not-working

### To Reproduce

`next dev` or `next build` on the reproduction repository.

### Describe the Bug

```
Failed to compile.

./node_modules/.pnpm/lucide-react@0.263.1_react@18.2.0/node_modules/lucide-react/dist/esm/lucide-react.mjs
Module not found: Can't resolve '/Users/joulev/dev/www/debug/node_modules/.pnpm/lucide-react@0.263.1_react@18.2.0/node_modules/lucide-react/dist/esm/icons/create-lucide'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./app/icons.tsx
./app/page.tsx
```

### Expected Behavior

The icon should work normally.

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

<sub>[NEXT-1495](https://linear.app/vercel/issue/NEXT-1495/createlucideicon-from-lucite-react-not-working-from-v13413-canary5)</sub>

### PR DESCRIPTION
**Pull request by chungweileong94**
### What?
Fixed missing modularize imports mapping for `createLucideIcon` from `lucide-react`.

Related PR: #53697

### Why?
None of the import mapping is covering `createLucideIcon`.

### How?
Add an import mapping rule for `createLucideIcon`.

Fixes #53605

### CODE REVIEW

File: packages/next/src/server/config.ts [hunk 1]
```diff
@@ -713,6 +713,9 @@ function assignDefaults(
           'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/sparkles!lucide-react',
         '(TextSelection|LucideTextSelection|TextSelectionIcon)':
           'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/text-select!lucide-react',
+        // Alias for createLucideIcon
+        '(createLucideIcon)':
+          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/createLucideIcon!lucide-react',
```
Comment (jguddas) [inline packages/next/src/server/config.ts:718]: Great catch, then we have nearly everything, the only thing that is missing now is `icons`.
```js
import * as index from './icons/index.js';
export { index as icons };
```


```suggestion
        // Alias for createLucideIcon
        '(createLucideIcon)':
          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/createLucideIcon!lucide-react',
        // Alias for icons
        '(icons)':
          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/index!lucide-react',
```

**Review by jguddas — APPROVED** [review]

File: packages/next/src/server/config.ts [hunk 2]
```diff
@@ -713,6 +713,12 @@ function assignDefaults(
...
+        // Alias for icons
+        '(icons)':
```
Comment (shuding) [inline packages/next/src/server/config.ts:720]:
```suggestion
        'icons':
```

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: ea808003706c4a537a2d97c7a4e136aa323b8878

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by shuding — DISMISSED** [review]

**Comment by chungweileong94** [comment]
I think the test failed for some other reason?

**Review by jguddas — APPROVED** [review]

**Comment by jd-carroll** [comment]
Can we please close this PR, it should NOT be merged. 
(Not that I don't appreciate the work that @chungweileong94 has done 😎 )

The issue is that this change does not actually fix anything. It continues the game of "whack-a-mole" with `lucide-react` everytime they release a new alias.

Lucide made _an explicit change_ to support Next.JS transforms:
https://github.com/lucide-icons/lucide/pull/1486

The fix we should merge is: #53789

We need to consolidate around that fix.

Please please please can we get this one closed and that one merged. Thank you ❤️ 

(@jguddas @shuding fyi)

**Comment by chungweileong94** [comment]
@jd-carroll Oh yeah of course, feel free to close it! 
Heard about the alias fix, and I definitely support that. This PR was created as a rough solution before the proper fix.

**Comment by jguddas** [comment]
> The fix we should merge is: https://github.com/vercel/next.js/pull/53789

This PR is still 100% valid, we only support aliases for icon names, `createLucideIcon` and `icons` still need to be hard-coded.

**Comment by jd-carroll** [comment]
> This PR is still 100% valid, we only support aliases for icon names, `createLucideIcon` and `icons` still need to be hard-coded.

Thank you for correcting me 😊

**Comment by jd-carroll** [comment]
Should we create a consolidated PR to make it easier for the Next team to wrap their heads around and merge?

**Pull request closed without merging**